pytorch/pytorch_lightning_ddp.py

```diff
@@ -123,21 +123,24 @@ def objective(trial: optuna.trial.Trial) -> float:
 
     model = LightningNet(dropout, output_dims)
     datamodule = FashionMNISTDataModule(data_dir=DIR, batch_size=BATCHSIZE)
+    callback = PyTorchLightningPruningCallback(trial, monitor="val_acc")
 
     trainer = pl.Trainer(
         logger=True,
         limit_val_batches=PERCENT_VALID_EXAMPLES,
         enable_checkpointing=False,
         max_epochs=EPOCHS,
-        gpus=-1 if torch.cuda.is_available() else None,
-        accelerator="ddp_cpu" if not torch.cuda.is_available() else None,
-        num_processes=os.cpu_count() if not torch.cuda.is_available() else None,
-        callbacks=[PyTorchLightningPruningCallback(trial, monitor="val_acc")],
+        accelerator="auto" if torch.cuda.is_available() else "cpu",
```

Comment:
I'm not sure, but does `accelerator="auto"` check the availability of CUDA in the background? If so, we can remove the if statement.

Reply:
Ah, I found that `ddp_spawn` was incompatible with `mps`. So, the if statement cannot be removed.

Reply:
Yes, I also tried `accelerator="auto"` without checking the availability of CUDA and got the incompatible error.

Reply:
Thanks!